docker-compose.yml

```
@@ -32,9 +34,22 @@ services:
     command: redis-server --save 20 1 --loglevel warning --requirepass '${REDIS_PASSWORD}'
     volumes:
       - cache:/data
+    profiles:
+      - dependencies
+    networks:
+      - recipe-app-network
+  application:
+    image: kuba1212/recipeapp-docker
+    env_file: .env
```

kudos to Arek for this:

```suggestion
  application:
    image: kuba1212/recipeapp-docker
    build: .
    env_file: .env
```

it will work like this:
1. if we have a `kuba1212/recipeapp-docker` image available then it will use this image
2. if not it will try to build it using Dockerfile

This way we can also use this docker-compose file locally for running the newest backend app in the docker too.

If we want to make sure that on the server we always fetch the image (because we don't have source files or Dockerfile) we should add `--no-build` flag, e.g. `docker-compose up --no-build`.

On the other hand, locally you can force to build docker image instead of pulling it by using `--build <service_name>`, e.g. `docker compose up --build <service_name>` (you should test / verify it though)